app/models/bulkrax/csv_entry.rb

```diff
@@ -249,7 +249,7 @@ def possible_collection_ids
       identifiers = []
       split_references = record[parent_field_mapping].split(/\s*[;|]\s*/)
       split_references.each do |c_reference|
-        matching_collection_entries = importerexporter.entries.select { |e| e.raw_metadata[work_identifier] == c_reference && e.is_a?(CsvCollectionEntry) }
+        matching_collection_entries = importerexporter.entries.select { |e| e.raw_metadata[source_identifier] == c_reference && e.is_a?(CsvCollectionEntry) }
```

Comment:
use the `source_identifier` value since that's what's on the csv and we're looking at the raw metadata.